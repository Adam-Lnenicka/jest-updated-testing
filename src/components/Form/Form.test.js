import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";
import { act } from "react-dom/test-utils";

test("submits correctly", async () => {
  const handleCompleteMock = jest.fn();
  const signUpAPIMock = jest.fn();

  jest.doMock("./_signUp", () => ({
    __esModule: true,
    default: signUpAPIMock,
  }));

  const Form = require("./Form").default;

  let wrapper;

  await act(async () => {
    wrapper = mount(<Form onComplete={handleCompleteMock} />);
    wrapper.find("input[name='fullName']").instance().value = "Lucas";
    wrapper.find("input[name='email']").instance().value = "lucas@example.com";
    wrapper.find("form").simulate("submit");
  });

  expect(signUpAPIMock).toHaveBeenCalledWith({
    email: "lucas@example.com",
    User: "test-author",
    fullName: "Lucas",
  });
  expect(handleCompleteMock).toHaveBeenCalled();
});

test("matches snapshot", async () => {
  const Form = require("./Form").default;

  let wrapper;

  await act(async () => {
    wrapper = mount(<Form />);
  });

  expect(toJSON(wrapper)).toMatchSnapshot();
});
