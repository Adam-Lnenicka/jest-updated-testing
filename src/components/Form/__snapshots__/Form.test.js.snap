// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<BasicForm>
  <div>
    <h1>
      Signup
    </h1>
    <Formik
      initialValues={
        Object {
          "email": "",
          "fullName": "",
        }
      }
      onSubmit={[Function]}
      validationSchema={
        ObjectSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_excludedEdges": Array [],
          "_nodes": Array [
            "email",
            "fullName",
          ],
          "_sortErrors": [Function],
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {},
          "fields": Object {
            "email": StringSchema {
              "_blacklist": ReferenceSet {
                "list": Set {},
                "refs": Map {},
              },
              "_blacklistError": undefined,
              "_mutate": undefined,
              "_typeError": [Function],
              "_whitelist": ReferenceSet {
                "list": Set {},
                "refs": Map {},
              },
              "_whitelistError": undefined,
              "conditions": Array [],
              "deps": Array [],
              "exclusiveTests": Object {
                "email": false,
                "required": true,
              },
              "spec": Object {
                "abortEarly": true,
                "nullable": false,
                "presence": "required",
                "recursive": true,
                "strict": false,
                "strip": false,
              },
              "tests": Array [
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
              "type": "string",
            },
            "fullName": StringSchema {
              "_blacklist": ReferenceSet {
                "list": Set {},
                "refs": Map {},
              },
              "_blacklistError": undefined,
              "_mutate": undefined,
              "_typeError": [Function],
              "_whitelist": ReferenceSet {
                "list": Set {},
                "refs": Map {},
              },
              "_whitelistError": undefined,
              "conditions": Array [],
              "deps": Array [],
              "exclusiveTests": Object {
                "max": true,
                "min": true,
                "required": true,
              },
              "spec": Object {
                "abortEarly": true,
                "nullable": false,
                "presence": "required",
                "recursive": true,
                "strict": false,
                "strip": false,
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
              "type": "string",
            },
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "optional",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
          "type": "object",
        }
      }
    >
      <Form>
        <form
          action="#"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <Field
            name="fullName"
          >
            <input
              name="fullName"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            />
          </Field>
          <Field
            name="email"
            type="email"
          >
            <input
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              type="email"
              value=""
            />
          </Field>
          <button
            type="submit"
          >
            Submit
          </button>
        </form>
      </Form>
    </Formik>
  </div>
</BasicForm>
`;
